### ISSUE TITLE
Action based rate limiting

### ISSUE THREAD

**Issue by rmarronnier** [luckyframework/lucky#1865]
To protect a web app against abusive connections / requests (sub DDOS scale) several layers can help (nginx, [middleware](https://github.com/defense-cr/defense)), but stumbling on [Rails future rate limiting ](https://github.com/rails/rails/pull/50490), I'm quite jealous of the elegant DX of this solution.

I could clearly see this working in Lucky :

```crystal
class SignIns::New < BrowserAction
  include Auth::RedirectSignedInUsers
  rate_limit to: 50, within: 10.seconds

  get "/sign_in" do
    html NewPage, operation: SignInUser.new
  end
end
``` 

I can't implement this right now, I'm just putting it out there as a starting point for discussion / inspiration.

**Comment by jwoertink** [comment]
Yeah, I dig it. I think rails has an easy way to handle it with having their key/value store built-in now. I wonder how this would scale with postgres backed. Or would this feature require redis to be added in?
